movement/movement.c

```diff
@@ -569,8 +580,13 @@ bool app_loop(void) {
     // if the watch face changed, we can't sleep because we need to update the display.
     if (movement_state.watch_face_changed) can_sleep = false;
 
-    // if the buzzer or the LED is on, we need to stay awake to keep the TCC running.
-    if (movement_state.is_buzzing || movement_state.light_ticks != -1) can_sleep = false;
+    // if we woke up for the buzzer, stay awake until it's finished.
+    if (woke_up_for_buzzer) {
+        while(watch_is_buzzer_or_led_enabled());
```

Comment:
I think this is the only line that gives me concerns... I take it this is hanging here until the buzzer is disabled in some other ISR? Is there a scenario where this could hang indefinitely?

Reply:
there shouldn't be — it spins here until the TCC is switched off, but only if it woke up for the buzzer. If it just woke up, the TCC can't be on for the LED.

the other tack we were thinking of was to spin while `is_beeping` is true, but i think that requires tagging more stuff with `volatile`, instead of probing the hardware state directly.